shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -131,15 +134,14 @@ class JSBackend {
     case Let(true, Var(name), Lam(args, body), expr) =>
       val letScope = scope.derive("Let", name :: Nil)
       val fn = {
-        val params = translateParams(args)
-        val bindings = name :: params.flatMap { _.bindings }
-        val fnScope = scope.derive("Function", bindings)
+        val params = translateParams(args)(letScope)
```

Comment:
This seems incorrect. Why are you translating the parameters in the `letScope` instead of the `fnScope`?